data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,243 @@
+//! This module contains data events
+
+use super::*;
+
+/// Trait for retreiving id from events
+pub trait IdTrait: Identifiable {
+    fn id(&self) -> &Self::Id;
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum AssetEvent {
+    Created(AssetId),
+    Deleted(AssetId),
+    Increased(AssetId),
+    Decreased(AssetId),
+    MetadataInserted(AssetId),
+    MetadataRemoved(AssetId),
+}
+
+impl Identifiable for AssetEvent {
+    type Id = AssetId;
+}
+
+impl IdTrait for AssetEvent {
+    fn id(&self) -> &AssetId {
+        match self {
+            Self::Created(id)
+            | Self::Deleted(id)
+            | Self::Increased(id)
+            | Self::Decreased(id)
+            | Self::MetadataInserted(id)
+            | Self::MetadataRemoved(id) => id,
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum AssetDefinitionEvent {
+    Created(AssetDefinitionId),
+    Deleted(AssetDefinitionId),
+    MetadataInserted(AssetDefinitionId),
+    MetadataRemoved(AssetDefinitionId),
+}
+
+impl Identifiable for AssetDefinitionEvent {
+    type Id = AssetDefinitionId;
+}
+
+impl IdTrait for AssetDefinitionEvent {
+    fn id(&self) -> &AssetDefinitionId {
+        match self {
+            Self::Created(id)
+            | Self::Deleted(id)
+            | Self::MetadataInserted(id)
+            | Self::MetadataRemoved(id) => id,
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum PeerEvent {
+    Created(PeerId),
```

Comment:
`Trusted` and `Untrusted` would be a better fit, but it's not that important to address in this particular PR.